src/storage/operations/head.rs

```diff
@@ -0,0 +1,306 @@
+
+        // Determine reading mode
+        // Default behavior (no flags) reads first 10 lines without confirmation.
+        let mode = match (lines, bytes) {
+            (Some(line_count), None) => HeadMode::Lines(line_count),
+            (None, Some(byte_count)) => HeadMode::Bytes(byte_count),
+            (None, None) => HeadMode::Lines(10), // Default 10 lines
+            (Some(_), Some(_)) => {
+                return Err(Error::InvalidArgument {
+                    message: "Cannot specify both lines and bytes options".to_string(),
+                });
+            }
+        };
+
+        // Read and display based on mode
+        match mode {
+            HeadMode::Lines(line_count) => {
+                self.head_by_lines(path, line_count).await?;
+            }
+            HeadMode::Bytes(byte_count) => {
+                self.head_by_bytes(path, byte_count).await?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Display the beginning of multiple files with minimal GNU head-like headers.
+    pub async fn head_many(
+        &self,
+        paths: &[String],
+        lines: Option<usize>,
+        bytes: Option<usize>,
+        quiet: bool,
+        verbose: bool,
+        force: bool,
+    ) -> Result<()> {
+        if quiet && verbose {
+            return Err(Error::InvalidArgument {
+                message: "Cannot specify both --quiet and --verbose".to_string(),
+            });
+        }
+
+        let should_show_header = |total: usize| -> bool {
+            if verbose {
+                return true;
+            }
+            if quiet {
+                return false;
+            }
+            total > 1
+        };
+
+        let total = paths.len();
+        for (idx, p) in paths.iter().enumerate() {
+            let show_header = should_show_header(total);
+            if show_header {
+                if idx > 0 {
+                    println!();
+                }
+                println!("==> {} <==", p);
+            }
+
+            if let Err(e) = self.read_and_display_head(p, lines, bytes, force).await {
+                eprintln!("{}", e);
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Read and display file content by lines using ranged, chunked reads.
+    async fn head_by_lines(&self, path: &str, max_lines: usize) -> Result<()> {
+        if max_lines == 0 {
+            return Ok(());
+        }
+
+        // Determine file size to clamp range end within object bounds.
+        let file_size = self
+            .operator
+            .stat(path)
+            .await
+            .map_err(|e| self.map_to_head_failed(path, e))?
+            .content_length();
+
+        // Use small ranged reads to avoid loading entire file.
+        const CHUNK_SIZE: u64 = 8192;
+        let stdout = io::stdout();
+        let mut handle = stdout.lock();
+
+        let mut next_offset: u64 = 0;
+        let mut lines_emitted: usize = 0;
+        let mut carry: Vec<u8> = Vec::new();
```

Comment:
The carry vectors can grow indefinitely, which is a memory leak risk for large files without line breaks